ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift

```diff
@@ -412,4 +442,54 @@ struct RoomTimelineItemFactory: RoomTimelineItemFactoryProtocol {
                               isEditable: false,
                               sender: eventTimelineItem.sender)
     }
+    
+    // MARK: - Reply details
+    
+    // swiftlint:disable:next cyclomatic_complexity
+    private func buildReplyToDetailsFrom(details: InReplyToDetails?) -> TimelineItemReplyDetails? {
+        guard let details else { return nil }
+        
+        switch details.event {
+        case .unavailable:
+            return .unavailable(eventID: details.eventId)
+        case .pending:
+            return .pending(eventID: details.eventId)
+        case let .ready(message, senderID, senderProfile):
+            let sender: TimelineItemSender
+            switch senderProfile {
+            case let .ready(displayName, _, avatarUrl):
+                sender = TimelineItemSender(id: senderID,
+                                            displayName: displayName,
+                                            avatarURL: avatarUrl.flatMap(URL.init(string:)))
+            default:
+                sender = TimelineItemSender(id: senderID,
+                                            displayName: nil,
+                                            avatarURL: nil)
+            }
+            
+            let replyContent: TimelineItemReplyContent
+            switch message.msgtype() {
+            case .audio(let content):
+                replyContent = .audio(buildAudioTimelineItemContent(content))
+            case .emote(let content):
+                replyContent = .emote(buildEmoteTimelineItemContent(senderDisplayName: sender.displayName, senderID: sender.id, messageContent: content))
+            case .file(let content):
+                replyContent = .file(buildFileTimelineItemContent(content))
+            case .image(let content):
+                replyContent = .image(buildImageTimelineItemContent(content))
+            case .notice(let content):
+                replyContent = .notice(buildNoticeTimelineItemContent(content))
+            case .text(let content):
+                replyContent = .text(buildTextTimelineItemContent(content))
+            case .video(let content):
+                replyContent = .video(buildVideoTimelineItemContent(content))
+            case .none:
+                replyContent = .text(TextRoomTimelineItemContent(body: "Unkown reply content"))
```

Comment:
Should this one be localised too, plus small typo.
```suggestion
                replyContent = .text(TextRoomTimelineItemContent(body: "Unknown reply content"))
```

Reply:
Okay for the typo but we shouldn't worry about the other one, I can't see any reason why that message type is optional. I'll ask the rust team.

Reply:
Actually, even better, just going to return nil here